instagram/src/Authentication/Authenticate.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import LoginPage from '../components/LoginPage/Loggin';
+
+const Authenticate = App =>
+    class extends React.Component {
+        constructor(props) {
+            super(props);
+                this.state = {
+                    loggedIn: false
+                };
+            }
+        
+        componentDidMount() {
+            if (!localStorage.getItem('user')) {
+                this.setState({ loggedIn: false });
+            } else {
+                this.setState({ loggedIn: true });
```

Comment:
An interesting alternative to this if else is something like this:
```javascript
this.setState({ loggedIn: !!localStorage.getItem('user') });
```
Here, the first NOT operator (`!`) coerces our null or string into a boolean, and then the next NOT makes it the opposite.